politeiad/backendv2/tstorebe/store/mysql/encrypt.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mysql
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/decred/politeia/util"
+	"github.com/marcopeereboom/sbox"
+)
+
+func (s *mysql) encrypt(ctx context.Context, tx *sql.Tx, data []byte) ([]byte, error) {
+	// Create a new nonce value
+	_, err := tx.ExecContext(ctx, "INSERT INTO nonce () VALUES ();")
```

Comment:
This is not at all obvious if it is a transaction.

Reply:
Its a `slq.Tx` type.  Unique nonces have also been verified via a test that can be run against a mariadb instance.

https://github.com/decred/politeia/pull/1180/files#diff-3b9ef4ced304f99d6d750cbfcdb9f31a6fa3ff108b822085f3574ffd0fdf56dfR1

Reply:
Just expand the comment. `// Create a new nonce value in an atomic database transaction.`

Reply:
Done

https://github.com/lukebp/politeia/blob/tlog/politeiad/backendv2/tstorebe/store/mysql/nonce.go#L15-L54

Reply:
Really dislike the mysql LAST_INSERT_ID dance but it is what it is.